specs/Deployment/MSIXPackages.md

```diff
@@ -0,0 +1,189 @@
+to allow the servicing event to occur and restarted upon its completion, with only a transient (and
+possibly unnoticed) interruption of service.
+
+Further, if/when the Main package is necessary, developers are encouraged to put as
+little as practical in the Main package and use the Framework package as much as possible. For
+example, if a StartupTask is needed you can declare the StartupTask in the Main package's
+`appxmanifest.xml` and provide an executable with a trivial 1% implementation e.g.
+
+```c
+int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, PWSTR pCmdLine, int nCmdShow)
+{
+    return MyFeatureWinMain(hInstance, hPrevInstance, pCmdLine, nCmdShow);
+}
+```
+
+and export `MyFeatureWinMain` from a DLL in the Framework package to get the implementation's
+other 99%.
+
+### Role - Singleton
+
+The Singleton package supplements the Main package, for when functionality is needed by a single
+process spanning all versions of Windows App SDK.
+
+### Dynamic Dependency Lifetime Manager (DDLM)
+
+The [Dynamic Dependency API](https://github.com/microsoft/WindowsAppSDK/blob/main/specs/dynamicdependencies/DynamicDependencies.md)
+resolves a package dependency to a specific framework package and make it available for a process,
+but Windows doesn't know the package is in use by this process. Lacking this "package in use"
+knowledge, Windows could service the package (remove, remediate, repair, ...) in negatively
+impactful ways. To prevent this (without servicing Windows) we need...
+
+1. ...a running process running with package identity...
+2. ...with an `appxmanifest.xml` declaring a `<PackageDependnecy>` on the framework package...
+3. ...running for as long as the framework package is used by the app.
+
+This is accomplished for the Microsoft.WindowsAppRuntime framework package via a Packaged COM
+out-of-process (OOP) server in a DDLM package, selected and managed via the Bootstrap API. See the
+[Dynamic Dependency spec](https://github.com/microsoft/WindowsAppSDK/blob/main/specs/dynamicdependencies/DynamicDependencies.md)
+for more details.
+
+## Naming
+
+Windows App SDK's MSIX packages use the following naming rules for package identity:
+
+* Name = Microsoft.WindowsAppRuntime[.SubName].<Major>.<Minor>[-VersionTag]
+* Publisher = "CN=Microsoft Corporation, O=Microsoft Corporation, L=Redmond, S=Washington, C=US"
+
+where
+
+* SubName -- optional package sub-name. Use to distinguish amongst the different packages
+* Major -- major version of the release, e.g. "1" for Windows App SDK 1.0
+* Minor -- minor version of the release, e.g. "0" for Windows App SDK 1.0
+* VersionTag -- optional version tag to distinguish amongst channels and revs of a channel, e.g. "-preview2" for Windows App SDK 1.0 Preview 2
+
+### Naming - SubName
+
+The following SubName values are used:
+
+| SubName | Package | Example |
+|-|-|-|
+| | Framework | Microsoft.WindowsAppRuntime.1.0-experimental1 |
```

Comment:
not exp1? char count keeps increasing

Reply:
Channel names are Stable, Preview and Experimental.
Their version tags are "" (none), "-preview" and "-experimental"
They also can have a numeric suffix e.g. "-experimental1".

That's also why we need the 'shortversiontag' e.g. -e1 == -experimental1.

And yes, names keep getting longer since 0.1 days. But so too do our requirements so it shouldn't be too surprising ;-)

I went through the exhaustive exercise to design a naming scheme that lets MSIX packages meet our versioning principles and policies while also working within the bounds of the available technology. That's also why DDLM packages have unusual names (but they also pose unusual requirements). Too many details not mentioned here but already well defined (been following those rules since 0.8-preview). Let me add that information.